packages/typescript/src/preflight.ts

```diff
@@ -0,0 +1,19 @@
+import { PluginContext } from 'rollup';
+import { ModuleKind } from 'typescript';
+
+import { TypeScriptConfig } from './options/tsconfig';
+
+const moduleError = `
+Rollup requires that TypeScript produces ES Modules. Unfortunately your configuration specifies a
+ "module" other than "esnext". Unless you know what you're doing, please change "module" to "esnext"
+ in the target tsconfig.json file or plugin options.`.replace(/\n/g, '');
+
+let undef;
+const validModules = [ModuleKind.ES2015, ModuleKind.ES2020, ModuleKind.ESNext, undef];
```

Comment:
@lukastaegert @NotWoods Note here that `ES6` isn't available in this context. There seems to be some weird fragmentation in the TypeScript source. Here's the code from `typescript.js` that define the available values:

```js
    (function (ModuleKind) {
        ModuleKind[ModuleKind["None"] = 0] = "None";
        ModuleKind[ModuleKind["CommonJS"] = 1] = "CommonJS";
        ModuleKind[ModuleKind["AMD"] = 2] = "AMD";
        ModuleKind[ModuleKind["UMD"] = 3] = "UMD";
        ModuleKind[ModuleKind["System"] = 4] = "System";
        // NOTE: ES module kinds should be contiguous to more easily check whether a module kind is *any* ES module kind.
        //       Non-ES module kinds should not come between ES2015 (the earliest ES module kind) and ESNext (the last ES
        //       module kind).
        ModuleKind[ModuleKind["ES2015"] = 5] = "ES2015";
        ModuleKind[ModuleKind["ES2020"] = 6] = "ES2020";
        ModuleKind[ModuleKind["ESNext"] = 99] = "ESNext";
    })(ModuleKind = ts.ModuleKind || (ts.ModuleKind = {}));
```

Reply:
ES6 likely gets transformed into ES2015 when the strings are converted into the equivalent enum values.